dss/storage/bundles.py

```diff
@@ -1,21 +1,44 @@
+import io
 from functools import lru_cache
 import json
 import typing
+import time
 
-from cloud_blobstore import BlobNotFoundError
+import cachetools
+from cloud_blobstore import BlobNotFoundError, BlobStore
+from cloud_blobstore.s3 import S3BlobStore
 
 from dss import Config, Replica
 from dss.storage.identifiers import DSS_BUNDLE_KEY_REGEX, DSS_BUNDLE_TOMBSTONE_REGEX, BundleTombstoneID, BundleFQID
 from dss.storage.blobstore import test_object_exists
+from dss.util import multipart_parallel_upload
+
+
+_bundle_manifest_cache = cachetools.LRUCache(maxsize=4)
 
 
-@lru_cache(maxsize=32)
 def get_bundle_manifest(
         uuid: str,
         replica: Replica,
         version: typing.Optional[str],
         *,
         bucket: typing.Optional[str] = None) -> typing.Optional[dict]:
+    key = BundleFQID(uuid, version).to_key()
```

Comment:
Typing is still ` -> typing.Optional[dict]` but it looks like this function doesn't return a dictionary anymore.